### ISSUE TITLE
PFblockerng compatible

### ISSUE THREAD

**Issue by chromecapone** [d43m0nhLInt3r/socialblocklists#16]
Can you make the snapchat and tiktok lists in a format compatible with pfblockerng? I have pfblockerng pulling your lists but its having hard time reading them. I *THINK* its because of the leading IP address and the expression at the bottom part of the tiktok file. I *THINK* pfblockerng likes a simple list of hosts one per line...but apparently it CAN read different formats...but for whatever reason, its having a hard time reading this format.

**Comment by d43m0nhLInt3r** [comment]
I will think about it but I dont promise anything.

**Comment by d43m0nhLInt3r** [comment]
Do you want any specific blocklist? I would port it and would make it avaiable in a new folder for pfblockerng

**Comment by chromecapone** [comment]
I was mainly interested in the tiktok one...  thanks! That would be great!

**Comment by gregf** [comment]
These lists work fine with pfblockerng, you must be doing something wrong if it's not. Have you done a force reload? Are the lists format set to auto?

**Comment by chromecapone** [comment]
Yeah, you could be right. Let me give it another try and I'll get back to
you! Thanks for your help

**Comment by chromecapone** [comment]
I added the feed again...and it downloads just fine (last time it wouldn't
process the file), but now it does.

However, I can still access tiktok...just by typing it into chrome.

**Comment by d43m0nhLInt3r** [comment]
Thats strange... I used the list for pi hole a while ago and it worked great. I tested it a few minutes ago with uBlock Origin and the page couldnt load. I never used PFblockerng so I cant tell you where the problem is.

**Comment by chromecapone** [comment]
Pfblockerng IS blocking SOME of the listed sites...but not everything in
your list...and I suspect it's the way the pfblockerng handles the wildcard
expressions at the bottom of your list.

(^|\.)muscdn\.com$

(^|\.)musical\.ly$

(^|\.)tiktok\.com$

(^|\.)tiktok\.org$

(^|\.)tiktokcdn\.com$

anyway, that's my GUESS.

**Comment by gregf** [comment]
Yeah, dns can't block regex. Neither can't the pihole (unless it goes in a separate regex file). You'll need to find a different list or create a small one of your own that blocks specific domains. I know of other lists, but I don't want to really advertise them here as I don't feel its fair to the dev.

**Comment by chromecapone** [comment]
I was able to grab the wildcard domains at the bottom of his list, enabled
TLD blocking in pfblockerng-devel, and added those domains to the
blacklist. (Enabling the TLD feature will block the subdomains as well).
Seems like everything is working perfectly now.. So, for me, it was a two
step process. I added d43m0nhLInt3r tiktok list... plus I added the domains
at the bottom of the list right into pfblockerng and everything seems to be
working now.  I agree with you about advertising the lists here. If there's
a way to PM me with the lists...I'd be interested in evaluating them.
Thanks!

**Comment by d43m0nhLInt3r** [comment]
Can you send me the list. I would create a list for pfblockerng.

> Yeah, dns can't block regex. Neither can't the pihole (unless it goes in a separate regex file). You'll need to find a different list or create a small one of your own that blocks specific domains. I know of other lists, but I don't want to really advertise them here as I don't feel its fair to the dev.

**Comment by d43m0nhLInt3r** [comment]
Sooo.. I just created a blacklist without regex. https://github.com/d43m0nhLInt3r/socialblocklists/blob/master/TikTok/tiktokblocklistWithoutRegex.txt

**Comment by chromecapone** [comment]
Thank you for creating that! You're awesome!

So, I decided I would like to try your new list (without regex).... I
removed everything in the blacklist and added a "without REGEX" feed.

I get the error message:

[ TikTokwithoutRegex ] Reload [ 08/28/20 16:15:32 ] . completed .
  IDN converted: [ can’t ] [ xn--cant-x96a ].
  Whitelist: github.com|
  ----------------------------------------------------------------------
  Orig.    Unique     # Dups     # White    # TOP1M    Final
  ----------------------------------------------------------------------
  3        2          0          1          0          1
  ----------------------------------------------------------------------
Tiktok works at this point.

So, I disabled that feed and re-enabled the original list (without adding
anything to the blacklist) ... and the original list is working perfectly
now... tiktok is BLOCKED.

I am definitely pulling the rest of my hair out with this one.

Thanks so much for your help!
